.github/workflows/ci.yaml

```diff
@@ -0,0 +1,69 @@
+name: colcon-core-ci
+
+on:
+  push:
+    branches: ['master']
+  pull_request:
+    branches: ['*']
+
+jobs:
+  pytest:
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        python: [3.6, 3.7, 3.8, 3.9]
+
+    name: colcon-core pytest
+    runs-on: ${{matrix.os}}
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{matrix.python}}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{matrix.python}}
+      - name: Install dependencies
+        run: |
+          python -m pip install -U pip setuptools
+          python -m pip install -U coloredlogs distlib EmPy mock pytest pytest-cov pytest-repeat pytest-rerunfailures
+          python -m pip install -U flake8-blind-except flake8-builtins flake8-class-newline flake8-comprehensions flake8-deprecated flake8-docstrings flake8-import-order flake8-quotes pep8-naming pylint scspell3k
+      - name: Run tests
+        run: python -m pytest --cov=colcon_core --cov-branch
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1
+
+  bootstrap:
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        python: [3.6, 3.7, 3.8, 3.9]
+
+    name: colcon-core bootstrap
+    runs-on: ${{matrix.os}}
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{matrix.python}}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{matrix.python}}
+      - name: Install dependencies
+        run: |
+          python -m pip install -U pip setuptools
+          python -m pip install -U coloredlogs distlib EmPy mock pytest pytest-cov pytest-repeat pytest-rerunfailures
+          python -m pip install -U flake8-blind-except flake8-builtins flake8-class-newline flake8-comprehensions flake8-deprecated flake8-docstrings flake8-import-order flake8-quotes pep8-naming pylint scspell3k
+      - name: Build and test
+        run: |
+          python bin/colcon build --build-base ../build --install-base ../install
+          python bin/colcon test --build-base ../build --install-base ../install
+      - name: Use the installed package
+        if: ${{runner.os != 'windows'}}
+        run: |
+          . ../install/local_setup.sh
+          colcon --help
```

Comment:
It might be worth following up sometime on why we can't enable this on Windows. If I set `shell: sh` on this step, it successfully calls `local_setup.sh`, but then it can't find the `colcon` executable.

Might actually be something we could fix - not sure how usable `sh` is on Windows.

Reply:
According to [the docs](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell) `bash` is supported on Windows and uses the bash binary from Git for Windows.

Reply:
`colcon-core` itself doesn't actually support bash (that's what [`colcon-bash`](https://github.com/colcon/colcon-bash) is for). Only `sh` and `bat` are available here.

Reply:
I wonder what it did when I specified `shell: sh` on Windows then? It must have used `bash` - there is no way the syntax used here is valid powershell or cmd.

Reply:
> `colcon-core` itself doesn't actually support bash (that's what [`colcon-bash`](https://github.com/colcon/colcon-bash) is for). Only `sh` and `bat` are available here.

Presumably though bash is compliant enough that the local_setup.sh setup script would work no?



> I wonder what it did when I specified `shell: sh` on Windows then? It must have used `bash` - there is no way the syntax used here is valid powershell or cmd.

That'd be my guess. Usually bash invoked as sh runs in a compatibility mode.

Reply:
I was able to get this to work, so now we should get coverage on running the bash hooks under Windows as part of the bootstrap tests.